File: src/main/java/com/chatsummary/bot/repository/DailySummaryRepository.java
package com.chatsummary.bot.repository;

import com.chatsummary.bot.model.DailySummary;
import java.time.Instant;
import java.util.List;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DailySummaryRepository extends MongoRepository<DailySummary, String> {

    List<DailySummary> findByChatIdAndTimestampAfter(long chatId, Instant since);

    void deleteByChatIdAndTimestampBefore(long chatId, Instant before);
}

File: src/main/kotlin/com/chatsummary/bot/repository/DailySummaryRepository.kt
package com.chatsummary.bot.repository

import com.chatsummary.bot.model.DailySummary
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.time.Instant

@Repository
interface DailySummaryRepository : MongoRepository<DailySummary, String> {
    fun findByChatIdAndTimestampAfter(chatId: Long, since: Instant): List<DailySummary>
    fun deleteByChatIdAndTimestampBefore(chatId: Long, before: Instant)
}
